EngineLayer/ClassicSearch/MiniClassicSearchEngine.cs

```diff
@@ -0,0 +1,230 @@
+            SpectralLibrary = spectralLibrary;
+            myCommonParameters = commonParameters;
+        }
+
+        /// <summary>
+        /// Searches all Ms2 scans in a 2 minute window around a peak apex for possible PSMs.
+        /// Psms are scored against a peptideWithSetModifications that acted as a donor in MBR.
+        /// Calculates traditional and spectral contrast angle scores
+        /// </summary>
+        /// <param name="donorPwsm"> Ms2 scans in window are searched for matches to this donor peptide</param>
+        /// <param name="peakApexRT"> The center of the 2 minute window where the search occurs</param>
+        /// <returns></returns>
+        public IEnumerable<PeptideSpectralMatch> SearchAroundPeak(PeptideWithSetModifications donorPwsm, double peakApexRT)
+        {
+            var targetFragmentsForEachDissociationType = new Dictionary<DissociationType, List<Product>>();
+
+            // check if we're supposed to autodetect dissociation type from the scan header or not
+            if (myCommonParameters.DissociationType == DissociationType.Autodetect)
+            {
+                foreach (var item in GlobalVariables.AllSupportedDissociationTypes.Where(p => p.Value != DissociationType.Autodetect))
+                {
+                    targetFragmentsForEachDissociationType.Add(item.Value, new List<Product>());
+                }
+            }
+            else
+            {
+                targetFragmentsForEachDissociationType.Add(myCommonParameters.DissociationType, new List<Product>());
+            }
+
+            // clear fragments from the last peptide
+            foreach (var fragmentSet in targetFragmentsForEachDissociationType)
+            {
+                fragmentSet.Value.Clear();
+            }
+
+            // score each scan that has an acceptable precursor mass
+            IEnumerable<ScanWithIndexAndNotchInfo> acceptableScans = GetAcceptableScans(donorPwsm.MonoisotopicMass, peakApexRT, SearchMode); // GetAcceptableScans is asynchronous, in case you care
+            if (!acceptableScans.Any())
+            {
+                return null;
+            }
+
+            List<PeptideSpectralMatch> acceptablePsms = new();
+            foreach (ScanWithIndexAndNotchInfo scan in acceptableScans)
+            {
+                var dissociationType = myCommonParameters.DissociationType == DissociationType.Autodetect ?
+                    scan.TheScan.TheScan.DissociationType.Value : myCommonParameters.DissociationType;
+
+                if (!targetFragmentsForEachDissociationType.TryGetValue(dissociationType, out var peptideTheorProducts))
+                {
+                    //TODO: print some kind of warning here. the scan header dissociation type was unknown
+                    continue;
+                }
+
+                // check if we've already generated theoretical fragments for this peptide+dissociation type
+                if (peptideTheorProducts.Count == 0)
```

Comment:
Because this is inside a foreach loop that iterates through scans, I believe it is still necessary. I may be wrong though. Are targetFragmentsForEachDissociationType dicts specific to one peptide?